anchor/client/src/lib.rs

```diff
@@ -750,93 +748,50 @@ async fn init_from_beacon_node<E: EthSpec>(
     Ok((genesis.genesis_time, genesis.genesis_validators_root))
 }
 
-async fn wait_for_genesis(
-    beacon_nodes: &BeaconNodeFallback<SystemTimeSlotClock>,
-    genesis_time: u64,
-) -> Result<(), String> {
-    let now = SystemTime::now()
-        .duration_since(UNIX_EPOCH)
-        .map_err(|e| format!("Unable to read system time: {e:?}"))?;
+async fn wait_for_genesis(genesis_time: u64) -> Result<(), String> {
+    let get_now = || {
+        SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .map_err(|e| format!("Unable to read system time: {e:?}"))
+    };
     let genesis_time = Duration::from_secs(genesis_time);
 
     // If the time now is less than (prior to) genesis, then delay until the
     // genesis instant.
     //
     // If the validator client starts before genesis, it will get errors from
     // the slot clock.
-    if now < genesis_time {
+    if get_now()? < genesis_time {
         info!(
-            seconds_to_wait = (genesis_time - now).as_secs(),
+            seconds_to_wait = (genesis_time - get_now()?).as_secs(),
```

Comment:
Can we return this result or zero if it's negative? Then we don't need the if/else, and we can have a log "Genesis occurred" inside the loop. Wdyt?